File: KickyBall.Application/ClientApp/src/controllers/game.controller.ts
import { HttpClient, HttpHandler } from "@angular/common/http";
import { Observable } from "rxjs";
import { Game } from "src/models/game.model";
import { Round } from "src/models/round.model";
import { GoalAttempt } from "src/models/goal-attempt.model";
import { RecordGoalAttemptRequest } from "src/requests/recordGoalAttemptRequest";

export class GameController {
    private http: HttpClient;
    private baseUrl: string;

    constructor(http: HttpClient){
        this.http = http;
        this.baseUrl = 'api/Game/';
    }

    GetCurrentGame(userId: number): Observable<Game>{
        return this.http.get<Game>(this.baseUrl + 'GetCurrentGame?userId=' + userId);
    }

    GetGameGoals(gameId: number): Observable<number>{
        return this.http.get<number>(this.baseUrl + 'GetGameGoals?gameId=' + gameId);
    }

    GetPracticeGoals(gameId: number): Observable<number>{
        return this.http.get<number>(this.baseUrl + 'GetPracticeGoals?gameId=' + gameId);
    }

    GetRoundGoals(roundId: number): Observable<number>{
        return this.http.get<number>(this.baseUrl + 'GetRoundGoals?roundId=' + roundId);
    }

    GetRouteIdsForGame(gameId: number, take: number): Observable<number[]>{
        return this.http.get<number[]>(this.baseUrl + 'GetRouteIdsForGame?gameId=' + gameId + '&take=' + take);
    }

    GetGoalAttemptNumberForRound(roundId: number, take: number): Observable<number>{
        return this.http.get<number>(`${this.baseUrl}GetGoalAttemptNumberForRound?roundId=${roundId}&take=${take}` );
    }

    CreateGame(game: Game): Observable<Game>{
        return this.http.post<Game>(this.baseUrl + 'CreateGame', game);
    }

    FinishGame(gameId: number): Observable<boolean>{
        return this.http.post<boolean>(this.baseUrl + 'FinishGame', gameId);
    }

    CreateRound(round: Round): Observable<Round>{
        return this.http.post<Round>(this.baseUrl + 'CreateRound', round);
    }

    FinishRound(roundId: number): Observable<boolean>{
        return this.http.post<boolean>(this.baseUrl + 'FinishRound', roundId);
    }

    RecordGoalAttempt(request: RecordGoalAttemptRequest): Observable<GoalAttempt>{
        return this.http.post<GoalAttempt>(this.baseUrl + 'RecordGoalAttempt', request);
    }
}

File: KickyBall.Application/Controllers/GameController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using KickyBall.BLL.Interfaces;
using KickyBall.BLL.Requests;
using KickyBall.BLL.Services;
using KickyBall.DAL;
using KickyBall.DAL.Models;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;

namespace KickyBall.Application.Controllers
{
    [ApiController]
    [Route("api/[controller]/[action]")]
    public class GameController : ControllerBase
    {
        private readonly ILogger<GameController> _logger;
        private readonly IGameService _service;

        public GameController(ILogger<GameController> logger, IGameService service)
        {
            _logger = logger;
            _service = service;
        }

        [HttpGet]
        public Game GetCurrentGame(int userId)
        {
            return _service.GetCurrentGame(userId);
        }

        [HttpGet]
        public int GetGameGoals(int gameId)
        {
            return _service.GetGameGoals(gameId);
        }

        [HttpGet]
        public int GetPracticeGoals(int gameId)
        {
            return _service.GetPracticeGoals(gameId);
        }

        [HttpGet]
        public int GetRoundGoals(int roundId)
        {
            return _service.GetRoundGoals(roundId);
        }

        [HttpGet]
        public List<int> GetEndPositionsForRound(int roundId)
        {
            return _service.GetEndPositionsForRound(roundId);
        }

        [HttpGet]
        public int GetGoalAttemptNumberForRound(int roundId)
        {
            return _service.GetGoalAttemptNumberForRound(roundId);
        }

        [HttpPost]
        public GoalAttempt RecordGoalAttempt(RecordGoalAttemptRequest request)
        {
            return _service.RecordGoalAttempt(request);
        }

        [HttpPost]
        public Round CreateRound(Round round)
        {
            return _service.CreateRound(round);
        }

        [HttpPost]
        public bool FinishRound([FromBody]int roundId)
        {
            return _service.FinishRound(roundId);
        }

        [HttpPost]
        public Game CreateGame(Game game)
        {
            return _service.CreateGame(game);
        }

        [HttpPost]
        public bool FinishGame([FromBody]int gameId)
        {
            return _service.FinishGame(gameId);
        }
    }
}
